docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,246 @@
+
+Don't be afraid to submit a PR, and most importantly, just get sucked in!
+
+## Development Pipeline
+
+As a broad overview, all changes to the Autograder start as an issue. Each 
+issue eventually gets a branch made to solve the issue. And each branch
+gets made in
+
+### 1. Issues
+
+All changes to the Autograder start as an issue. Issues help us track what 
+needs to be done and who's working on what. You can pick an already 
+existing issue to work on, or you can write your own. 
+
+> [!TIP]
+> Ideally, you should first work on issues in the `On Deck` column of the 
+[TA Project Board](https://github.com/orgs/softwareconstruction240/projects/1/views/9),
+but you are welcome to pick any unassigned issue
+
+#### Creating an Issue
+- Create the issue from the TA Projects Board (linked above)
+  - Click "add item" at the bottom of the `Todo` list
+  - Type `#autograder` to connect the issue to the Autograder
+  - Type a title and hit enter
+- Use a clear, descriptive title that uses one of these prefixes:
+  - `Backend:` server side changes
+  - `Frontend:` client (web) side changes
+  - `Fullstack:` for issues that require front and backend changes
+  - `Docs:` changes to documentation (either markdown files or JSDocs/JavaDocs)
+  - `Tests:` adding or modifying tests
+  - `Dev:` changes to enhance the Autograder development experience
+- Provide enough context so others can understand the purpose
+- Add labels as appropriate ([Here's the list](https://github.com/softwareconstruction240/autograder/issues/labels))
+- Link related issues
+- Issues should be narrow in scope
+
+#### Working on an Issue
+- Assign the issue to yourself before beginning work (this is how we know 
+  who is working on what, and gives you a sort of ownership over the issue)
+- Create a new branch ([more info here](#2-branches))
+- If you find a new issue, file it separately (don't mix concerns)
+- Leave a comment from time to time with updates on your progress
+
+### 2. Branches
+Branches in git allow us to all work on separate things in the same codebase.
+You will do all your work in a branch.
+
+#### Creating a branch
+Create the branch directly from the issue so your work is linked.
+
+- Always branch from `main` 
+  - (exceptions exist if you know what you're doing)
+- Use descriptive names that reflect what you're working on:
+    - `add-late-submission`
+    - `admin-display-verification-status`
+    - `server-communicator`
+    - `extract-service-logic-from-controllers`
+- Use kebab-case (lowercase with hyphens)
+- Keep names concise but clear
+- One branch per issue (don't mix different features/fixes)
+
+### 3. Making Changes
+Now that you have a branch, you can start to code! Hurray! 🎉
+Here are important guidelines to follow:
+
+- Review and follow the Style Guide (_Coming Soon_)
+- Write tests for your code. All new features and functionality must have tests demonstrating correctness
```

Comment:
(Half sarcastically) Might there be situations where this wouldn't apply? Might this be better worded with _should_ rather than _must_?

Reply:
I agree with @19mdavenport. In general, I don't like saying absolute statements and telling people what to do. I prefer to recommend good approaches and recognize that all kinds of unique cases come up. I prefer to assume that the people working on this project are smart, ambitious, capable TAs who want to make a difference and learn stuff along the way.

Additionally, I don't feel like there are enough untested portions of our current code base that we can't exactly demonstrate this principle by example. I absolutely agree on the vision of having strong testing coverage over our codebase.

Reply:
I'll go ahead and change it to should. I was just following the list of things @frozenfrank put on the issue for this PR.